lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -487,26 +489,34 @@ getWallet
     :: forall ctx s t k n.
         ( s ~ SeqState n k
         , ctx ~ ApiLayer s t k
+        , WalletKey k
         )
     => ctx
     -> ApiT WalletId
     -> Handler ApiWallet
-getWallet ctx wid =
-    fst <$> getWalletWithCreationTime mkApiWallet ctx wid
+getWallet ctx (ApiT wid) = do
+    reward <- liftHandler $ withWorkerCtx ctx wid (throwE . ErrFetchRewardsNoSuchWallet) $
+            \wrk -> W.fetchRewardBalance @_ @s @k @t wrk wid
+    fst <$> getWalletWithCreationTime (mkApiWallet reward) ctx (ApiT wid)
 
 listWallets
     :: forall ctx s t k n.
         ( s ~ SeqState n k
         , ctx ~ ApiLayer s t k
+        , WalletKey k
         )
     => ctx
     -> Handler [ApiWallet]
 listWallets ctx = do
     wids <- liftIO $ Registry.keys re
     fmap fst . sortOn snd <$>
-        mapM (getWalletWithCreationTime mkApiWallet ctx) (ApiT <$> wids)
+        mapM mkWallet wids
```

Comment:
**[suggestion]**

rewrite to `mapM (getWallet ctx) (ApiT<$> wids)` 

**Why**

Avoid logic duplication in `mkWallet`

Reply:
Unfortunately doesn't work since `listWallets` needs to sort on the creation time in `(ApiWallet, UTCTime)`, while `getWallet` only can return `ApiWallet`.

Maybe we can rewrite some of the `getWalletWithCreationTime` and `mkWallet`-indirection, however…

Reply:
06b04a5

I removed `mkApiWallet` and `mkApiByronWallet` which were arguments to `getWalletWithCreationTime`.

I merged `getWalletWithCreationTime` with `mkApiWallet` and added `getByronWalletWithCreationTime`.

This way there is less indirection.